elastic/kibana
# [Monitor management] Show public beta fair usage

## Summary
Fixes https://github.com/elastic/uptime/issues/466
Adds a callout for async Synthetics Service errors.

This callout will appear whenever errors are reported from the Synthetics Service during the sync monitors task. Errors are collected per location, with the reason and status displayed for each location. This allows the error to be generic enough for fair use, as well as other types of async errors.

<img width="1222" alt="Screen Shot 2022-03-29 at 2 09 05 PM" src="https://user-images.githubusercontent.com/11356435/160684491-3bf1d327-7e46-473f-bcfe-5e115b12c5f4.png">


## Testing
1. Connect to the Synthetics service via basic auth. The basic kibana.dev.yml config you'll need to do so is the following: https://p.elstc.co/paste/hsIRPa1C#yeHj2i3ILtcswV3k0d1JvIa7ACoQ3h2-a1DVPDXCv/q. You'll also likely want to set your sync interval to a smaller number to see test results more quickly. Set `xpack.uptime.service.syncInterval: 1m` in `kibana.dev.yml`.
2. Navigate to Monitor Management and add a single monitor.
3. In your Kibana.yml file, change `xpack.uptime.service.password` so that the authorization is no longer valid.
4. Navigate back to the Monitor Management list. Observe the sync error callout
5. Attempt to dismiss the callout. Verify that the callout is dismissed

Note: While this is not specifically testing fair use, this feature collects all possible sync errors.
